### PR TITLE
Implement drag and drop feature for the FileList

### PR DESCRIPTION
**Pull request by BohdanDudkaTD**
This PR implements logic to drag and drop files from the OS of the user to the list of files.
PR adds `StorageFile` - model of the file which is stored on the server. It might be a simple file or folder with files.
Also, it adds `FileListItem` - component which represents list item of the `FileList` component. Each `FileListItem` has its handlers for the drag and drop logic.
Besides, that handlers of drag and drop logic were added to the `FileList` component.

Result of the ESLint execution:
image

### CODE REVIEW

**Comment by BohdanDudkaTD** [comment]
@dkalinin, PTAL

File: app/store-configuration/action/index.js [hunk 1]
```diff
@@ -9,8 +9,20 @@ import {
 } from '../mutation/mutation-names';
 
 export default {
-  [ADD_ITEM_ACTION](context, payload) {
-    context.commit(ADD_ITEM_MUTATION, payload);
+  [ADD_ITEM_ACTION](context, currentFolderId, folderId, files) {
+    context.commit(CHANGE_ARE_FILES_LOADING_MUTATION, true);
```
Comment (dkalinin) [inline app/store-configuration/action/index.js]: SET_FILES_LOADING

File: app/view/component/file-list-item/index.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+ * Component which represents item of the the of items.
+ */
+export class FileListItem {
+  /**
+   * Constructs list item by rendering spans with details of the file and buttons to operate
+   * this item.
+   *
+   * @param {HTMLElement} parent -  element in which this component should be rendered
+   * (it might be an ol or an ul).
+   * @param {StorageFile} item - the instance of file model.
```
Comment (dkalinin) [inline app/view/component/file-list-item/index.js]: item => file

File: app/view/component/file-list/index.js [hunk 3]
```diff
@@ -33,25 +37,113 @@ export class FileList extends ConfigurableComponent {
...
-        const deleteButton = new Button(listItem, 'Delete');
-
-        deleteButton.onClick(() => {
-          deleteButton.disableButton();
+        listItem.onDeleteButtonClick(() => {
           super.dispatch(REMOVE_ITEM_ACTION, item.id);
         });
 
-        this._fileList.appendChild(listItem);
+        if (item.type === 'folder') {
```
Comment (dkalinin) [inline app/view/component/file-list/index.js]: If item.type is file nothing happens so there's no need to add event listeners in FileItem component.

File: app/view/component/file-list-item/index.js [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+  /**
+   * Removes HTML class from the list element.
+   *
+   * @param {string} className - name of the class which should be removed.
+   * @public
+   */
+  removeClass(className) {
+    this._dragCounter--;
+    if (this._dragCounter === 0) {
```
Comment (dkalinin) [inline app/view/component/file-list-item/index.js]: Adding or removing classes is not related to drag events. Please split this logic.

**Comment by BohdanDudkaTD** [comment]
@dkalinin, PTAL, fixed issues:
- renamed variables;
- logic to set/remove classes of 
HTML element was separated from the drag events logic;
- `DragAndDrop` component was added as a parent class for all components, which supports drag and drop events. Now each component sets event listeners only when it needs it;

File: app/model/storage-file/index.js [hunk 5]
```diff
@@ -19,58 +19,4 @@ export class StorageFile {
     this._type = type;
     this._fileType = fileType;
```
Comment (dkalinin) [inline app/model/storage-file/index.js]: Fields should be public.

File: app/view/component/file-list/index.js [hunk 6]
```diff
@@ -86,63 +64,26 @@ export class FileList extends ConfigurableComponent {
...
    *
    * @param {Event} event - drop event.
-   * @private
    */
   _onDrop(event) {
     event.preventDefault();
     event.stopPropagation();
-    this._dragCounter--;
+    this.onDragLeave(event, this._fileList);
     this._fileList.classList.remove('dragover');
```
Comment (dkalinin) [inline app/view/component/file-list/index.js]: Please remove this line.

**Pull request merged**